src/open_producten/producttypen/serializers/prijs.py

```diff
@@ -0,0 +1,99 @@
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework import serializers
+
+from ..models import Prijs, PrijsOptie, ProductType
+
+
+class PrijsOptieSerializer(serializers.ModelSerializer):
+    id = serializers.UUIDField(required=False)
+
+    class Meta:
+        model = PrijsOptie
+        exclude = ("prijs",)
+
+
+class PrijsSerializer(serializers.ModelSerializer):
+    prijsopties = PrijsOptieSerializer(many=True, default=[])
+    product_type_id = serializers.PrimaryKeyRelatedField(
+        source="product_type", queryset=ProductType.objects.all()
+    )
+
+    class Meta:
+        model = Prijs
+        exclude = ("product_type",)
+
+    def validate_prijsopties(self, opties: list[PrijsOptie]) -> list[PrijsOptie]:
+        if len(opties) == 0:
+            raise serializers.ValidationError(_("Er is minimaal één optie vereist."))
+        return opties
+
+    @transaction.atomic()
+    def create(self, validated_data):
+        prijsopties = validated_data.pop("prijsopties")
+        product_type = validated_data.pop("product_type")
+
+        prijs = Prijs.objects.create(**validated_data, product_type=product_type)
+
+        for optie in prijsopties:
+            PrijsOptie.objects.create(prijs=prijs, **optie)
+
+        return prijs
+
+    @transaction.atomic()
+    def update(self, instance, validated_data):
```

Comment:
This one is just confusing TBH.

Reply:
I think first you should check for duplicate IDS, then do the update/create stuff


``` python
if opties is not None:
    
    optie_errors = has duplicate

    if optie_errors:
        raise ValidationError
    else:
        # do the update logic

    

```

Reply:
Also their is the object not existing thing . 

IDK, I really dont like this endpoint

Reply:
moved validation to `PrijsOptieValidator`